service/service.go

```diff
@@ -95,7 +95,7 @@ func (svc *Service) Run(ctx context.Context, buildTime, gitCommit, version strin
 	}
 
 	// Get graphDB connection for observation store
-	if !svc.config.EnablePrivateEndpoints {
+	if !svc.config.EnablePrivateEndpoints || svc.config.DisableGraphDBDependency {
```

Comment:
Not sure on this logic.  If 
```
EnablePrivateEndpoints == true && svc.config.DisableGraphDBDependency == true
```
Then the 1st flag is now dependant on & can be overridden by the 2nd which could lead to some unexpected / misleading behaviour.

I appreciate what you're trying to do here but `EnablePrivateEndpoints` is used by the app to determine if we should enable access to sensitive data and I feel it that should be the only way to configure that decision. 

I appreciate I am being picky here but I'm concerned that this changes actually changes the flag from operating in a binary way - there is now a 3rd possible state.

Reply:
I'm wondering if there is an alternative solution to solve this problem.

Reply:
So my reasoning is - there's no way to add access to the GraphDB without explicitly enabling private endpoints still, and in that state with all things left unchanged, everything works exactly as it does already. 
It only gives the option to then disable one of those options enabled by the private endpoints, and it has to be explicitly set.

Reply:
The logic looks sound.
I've stepped through it with debugger and run it with extra printf's when running in command line and i see GraphDB not being used as expected when new flag is set false.

Reply:
Update: I've found an alternative solution to the particular problem I was trying to solve, but I feel some people may still find this a useful option to have - I'm open to either closing this PR or going ahead with it. Opinions?